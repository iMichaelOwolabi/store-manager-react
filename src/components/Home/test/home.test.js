import React from 'react';
import { shallow, mount } from 'enzyme';
import { Login } from '..';

describe('Test the Login Component', () => {
  let wrapper = shallow(<Login />);
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should change the state after change the input value', () => {
    const newValue = 'YoungM';
    wrapper.find('input[type="text"]').simulate('change', {
      target: { name: newValue }
    });
    wrapper.update();
    expect(wrapper.find('input[type="text"]').prop('name')).toContain('username');
    expect(wrapper.find('input[type="password"]').prop('name')).toContain('password');
  });

  it('should simulate on submit event ', () => {
    const props = {
      login: () => {}
    };
    wrapper = mount(<Login {...props} />);
    wrapper.find('form').simulate('submit', {
      preventDefault: jest.fn()
    });
    expect(wrapper.find('form')).toBeDefined();
  });
});
