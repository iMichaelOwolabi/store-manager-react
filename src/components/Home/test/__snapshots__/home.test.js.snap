// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the 404 Component should match snapshot 1`] = `
<h1>
  Welcome to the store manager
</h1>
`;
